### PR TITLE
'home work decorator'

### PR DESCRIPTION
**Pull request by FlyOn21**

### CODE REVIEW

File: hw_decorator.py [hunk 1]
```diff
@@ -1,15 +1,111 @@
...
+        return result_func
+
+    return wrapper
+
+
+@division_by_one_hundred
+def random_choice_number():
+    number = random.choice(range(100, 300))
+    return number
+
```
Comment (katevoskresenska) [inline hw_decorator.py:27]: If I am not mistaken, the task was to check if 100 can be divided by the number without remainder of the division

File: hw_decorator.py [hunk 2]
```diff
@@ -1,15 +1,111 @@
...
 # Если это int, тогда выполнить функцию и вывести результат, если это str(),
 # тогда зарейзить ошибку ValueError (raise ValueError(“string type is not supported”))
+def check_given_data(func):
+    """Decorator pre-checks the arguments passed to the function"""
+
+    @wraps(func)
+    def wrapper(*args, **kwargs):
+        given_data = [arg for arg in args if arg is not None] + \
+                     [value for key, value in kwargs.items() if value is not None]
+        if bool(given_data) is not True:
```
Comment (katevoskresenska) [inline hw_decorator.py]: if not given_data:

File: hw_decorator.py (hunk 1 above)
Comment (FlyOn21) [reply hw_decorator.py:27]: yes, I misunderstood the task, I will redo it

File: hw_decorator.py [hunk 3]
```diff
@@ -1,15 +1,110 @@
...
+
+def caching_functions_work(funk):
+    """Decorator impemented process caching the results of executing the wrapped function"""
+
+    @wraps(funk)
+    def wrapper(*args, **kwargs):
+        given_args = ''.join([str(arg) for arg in args if arg is not None])
+        given_kwargs_step_1 = sorted([(key, value) for key, value in kwargs.items() if value is not None])
+        given_kwargs_step_2 = ''.join([str(char) for unit in given_kwargs_step_1 for char in unit])
+        given_data = given_args + given_kwargs_step_2
```
Comment (mmykhailenko) [inline hw_decorator.py:72]: I think all these variables could be removed. Work directly with `args, kwargs`, no need to sort them or do any kind of staff.

File: hw_decorator.py [hunk 4]
```diff
@@ -1,15 +1,110 @@
...
 # ЗАДАЧА-2
 # Написать декоратор который будет выполнять предпроверку типа аргумента который передается в вашу функцию.
 # Если это int, тогда выполнить функцию и вывести результат, если это str(),
 # тогда зарейзить ошибку ValueError (raise ValueError(“string type is not supported”))
+def check_given_data(func):
+    """Decorator pre-checks the arguments passed to the function"""
+
+    @wraps(func)
+    def wrapper(*args, **kwargs):
+        given_data = [arg for arg in args if arg is not None] + \
```
Comment (mmykhailenko) [inline hw_decorator.py:38]: no need for these actions with` args, kwargs`

**Pull request closed without merging**